core/gsb-api/src/api.rs

```diff
@@ -0,0 +1,936 @@
+                    .await
+            }
+        );
+
+        ws_res.unwrap();
+        let gsb_res = gsb_res.unwrap().unwrap();
+        assert_eq!(gsb_res.content, vec![7; PAYLOAD_LEN]);
+
+        verify_delete_service(&mut api, &service_addr).await;
+    }
+
+    #[actix_web::test]
+    async fn error_payload_test() {
+        let mut api = dummy_api();
+
+        let (bind_req, service_addr) = bind_get_chunk_service_req(&mut api);
+        let body =
+            verify_bind_service_response(bind_req, vec!["GetChunk".to_string()], &service_addr)
+                .await;
+
+        let services_path = body.listen.unwrap().links.unwrap().messages;
+        let mut ws_frames = api.ws_at(&services_path).await.unwrap();
+
+        let gsb_endpoint = ya_service_bus::typed::service(&service_addr);
+        const TEST_ERROR_MESSAGE: &str = "test error msg";
+        let (gsb_res, ws_res) = tokio::join!(
+            async {
+                let msg = GetChunk {
+                    offset: u64::MIN,
+                    size: PAYLOAD_LEN as u64,
+                };
+                gsb_endpoint.call(msg).await
+            },
+            async {
+                let ws_req = ws_frames.try_next().await;
+                assert!(ws_req.is_ok());
+                let ws_req = ws_req.unwrap();
+                let ws_req = ws_req.unwrap();
+                let ws_req = match ws_req {
+                    Frame::Binary(ws_req) => {
+                        flexbuffers::from_slice::<TestWsRequest<GetChunk>>(&ws_req).unwrap()
+                    }
+                    msg => panic!("Not expected msg: {:?}", msg),
+                };
+                let id = ws_req.id;
+                let ws_res = serde_json::json!({
+                    "id": id,
+                    "error": {
+                    "InternalError": TEST_ERROR_MESSAGE
+                    }
+                });
+                let ws_res = flexbuffers::to_vec(ws_res).unwrap();
+                ws_frames
+                    .send(ws::Message::Binary(Bytes::from(ws_res)))
+                    .await
+            }
+        );
+
+        ws_res.unwrap();
+        let gsb_res = gsb_res.unwrap();
+        assert!(gsb_res.is_err());
+        let gsb_err = gsb_res.err().unwrap();
+        let expected_err =
+            ya_core_model::gftp::Error::InternalError(TEST_ERROR_MESSAGE.to_string());
+        match gsb_err {
+            ya_core_model::gftp::Error::InternalError(msg) => assert_eq!(msg, TEST_ERROR_MESSAGE),
+            other => panic!("Expected {:?} but got {:?}", expected_err, other),
+        }
+
+        verify_delete_service(&mut api, &service_addr).await;
+    }
+
+    #[test_case(r#"{}"#, Frame::Close(Some(CloseReason { 
+        code: CloseCode::Policy,
+        description: Some("Failed to read response. Err: Missing root map. Err: Empty map".to_string()) })); 
+        "Close when empty"
+    )]
+    //TODO Why None here
```

Comment:
Not sure what this todo means

Reply:
It is about Close reason message. For some reason is empty. Probably some typo I fail to notice. Not critical, because it is only a message.